lib/samples/apply_style_to_wms_layer/apply_style_to_wms_layer.dart

```diff
@@ -0,0 +1,165 @@
+// limitations under the License.
+//
+
+import 'package:arcgis_maps/arcgis_maps.dart';
+import 'package:arcgis_maps_sdk_flutter_samples/utils/sample_state_support.dart';
+import 'package:flutter/material.dart';
+
+class ApplyStyleToWmsLayer extends StatefulWidget {
+  const ApplyStyleToWmsLayer({super.key});
+
+  @override
+  State<ApplyStyleToWmsLayer> createState() => _ApplyStyleToWmsLayerState();
+}
+
+class _ApplyStyleToWmsLayerState extends State<ApplyStyleToWmsLayer>
+    with SampleStateSupport {
+  // Create a controller for the map view.
+  final _mapViewController = ArcGISMapView.createController();
+
+  // A flag for when the map view is ready and controls can be used.
+  var _ready = false;
+
+  // Hold a reference to the layer to enable re-styling.
+  late WmsLayer _wmsLayer;
+
+  // String array to store the styles.
+  final _stylesTitles = [
+    'Default',
+    'Contrast stretch',
+  ];
+
+  // Create variable for holding sublayer style.
+  String? _selectedStyle;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        top: false,
+        child: Stack(
+          children: [
+            Column(
+              children: [
+                Expanded(
+                  // Add a map view to the widget tree and set a controller.
+                  child: ArcGISMapView(
+                    controllerProvider: () => _mapViewController,
+                    onMapViewReady: onMapViewReady,
+                  ),
+                ),
+                // Build the bottom menu.
+                buildBottomMenu(),
+              ],
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            Visibility(
+              visible: !_ready,
+              child: const SizedBox.expand(
+                child: ColoredBox(
+                  color: Colors.white30,
+                  child: Center(child: CircularProgressIndicator()),
+                ),
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget buildBottomMenu() {
+    return Center(
+      // A drop down button for selecting style.
+      child: DropdownButton(
+        alignment: Alignment.center,
+        hint: const Text(
+          'Choose a style',
+          style: TextStyle(
+            color: Colors.deepPurple,
+          ),
+        ),
+        icon: const Icon(Icons.arrow_drop_down),
+        iconEnabledColor: Colors.deepPurple,
+        iconDisabledColor: Colors.grey,
+        style: const TextStyle(color: Colors.deepPurple),
+        value: _selectedStyle,
+        items: _stylesTitles.map((items) {
+          return DropdownMenuItem(
+            value: items,
+            child: Text(items),
+          );
+        }).toList(),
+        onChanged: (style) {
+          if (style != null) {
+            changeStyle(style);
+          }
+        },
+      ),
+    );
+  }
+
+  void changeStyle(String style) async {
```

Comment:
this method doesn't need to be `async`

Reply:
I used `await` to load the `_wmsLayer`:
```
// Wait for the layer to load.
    await _wmsLayer.load();
```

Reply:
`await _wmsLayer.load()` is down in `onMapViewReady`. Here in `changeStyle`, there are no `await` calls.

Reply:
oh fact, my bad!